drv/sidecar-mainboard-controller-api/src/tofino2.rs

```diff
@@ -133,12 +150,94 @@ impl Sequencer {
         }
     }
 
+    pub fn pcie_hotplug_ctrl(&self) -> Result<u8, FpgaError> {
+        self.fpga.read(Addr::PCIE_HOTPLUG_CTRL)
+    }
+
+    pub fn write_pcie_hotplug_ctrl(
+        &self,
+        op: WriteOp,
+        value: u8,
+    ) -> Result<(), FpgaError> {
+        self.fpga.write(op, Addr::PCIE_HOTPLUG_CTRL, value)
+    }
+
+    pub fn set_pcie_present(&self, present: bool) -> Result<(), FpgaError> {
+        self.write_pcie_hotplug_ctrl(
+            set_or_clear(present),
+            Reg::PCIE_HOTPLUG_CTRL::PRESENT,
+        )
+    }
+
+    pub fn set_pcie_power_fault(&self, fault: bool) -> Result<(), FpgaError> {
+        self.write_pcie_hotplug_ctrl(
+            set_or_clear(fault),
+            Reg::PCIE_HOTPLUG_CTRL::POWER_FAULT,
+        )
+    }
+
+    pub fn set_pcie_alert(&self, alert: bool) -> Result<(), FpgaError> {
+        self.write_pcie_hotplug_ctrl(
+            set_or_clear(alert),
+            Reg::PCIE_HOTPLUG_CTRL::ALERT,
+        )
+    }
+
+    pub fn pcie_reset(&self) -> Result<TofinoPcieReset, FpgaError> {
+        let ctrl = self.pcie_hotplug_ctrl()?;
+        match (
+            ctrl & Reg::PCIE_HOTPLUG_CTRL::RESET,
+            ctrl & Reg::PCIE_HOTPLUG_CTRL::OVERRIDE_HOST_RESET,
+        ) {
+            (_, 0) => Ok(TofinoPcieReset::HostControl),
+            (0, Reg::PCIE_HOTPLUG_CTRL::OVERRIDE_HOST_RESET) => {
+                Ok(TofinoPcieReset::Deasserted)
+            }
+            (
+                Reg::PCIE_HOTPLUG_CTRL::RESET,
+                Reg::PCIE_HOTPLUG_CTRL::OVERRIDE_HOST_RESET,
+            ) => Ok(TofinoPcieReset::Asserted),
+            _ => Err(FpgaError::InvalidValue),
+        }
```

Comment:
This is a little awkward / error-prone, because you have to make sure the bitflags in the tuples are lined up correctly.

What about explicitly extracting the bit flags, then matching on bools?
```suggestion
        let reset = (ctrl & Reg::PCIE_HOTPLUG_CTRL::RESET) != 0;
        let override_host_reset =
            (ctrl & Reg::PCIE_HOTPLUG_CTRL::OVERRIDE_HOST_RESET) != 0;
        match (reset, override_host_reset) {
            (_, false) => Ok(TofinoPcieReset::HostControl),
            (false, true) => Ok(TofinoPcieReset::Deasserted),
            (true, true) => Ok(TofinoPcieReset::Asserted),
        }
```